### ISSUE TITLE
[Feature] Favouriting repositories and sorting priority

### ISSUE THREAD

**Issue by LefsFlarey** [gogits/gogs#3657]
**Disclaimer:** I'm not sure if this has been requested before but I couldn't find any relevant issues like this one.
**TL;DR:** Ability to favourite issues and have them appear at the top of the user's repository list.
# The Problem

Having too many repositories causes the repository list to be clogged beyond reason, finding an important single repository becomes tedious.
# The Proposed Solution

Some work can be done to allow users to 'star' or 'favourite' their more important repositories to make them appear at the top of their repository list. This may include a creation of a new database table, particularly the `user_repo_favorite` table or something similar.
## user_repo_favorite

| user_id | repo_id |
| --- | --- |
| 1 | 1 |
| 1 | 2 |
| 2 | 1 |
| 1 | 3 |

This table can have a foreign relation with both the `user` and `repository` tables.
# So, what do you think?

**Comment by Unknwon** [comment]
Isn't this same as **star** a repository? (though we're missing the star list page)
